### PR TITLE
Fix for issue #27: smart plugin returns error if nothing is collected

### PR DESCRIPTION
**Pull request by obourdon**
Subject says it all

will wait for TravisCI to run tests before potentially submitting new revision as tests execution is system dependant

### CODE REVIEW

**Comment by obourdon** [comment]
cc: [@intelsdi-x/plugin-maintainers](https://github.com/orgs/intelsdi-x/teams/plugin-maintainers)

Could you please review and merge

File: smart/plugin_test.go [hunk 1]
```diff
@@ -279,7 +279,7 @@ func TestCollectMetrics(t *testing.T) {
 
 			ReadSmartData = func(device string,
 				sysutilProvider SysutilProvider) (*SmartValues, error) {
-				return &SmartValues{}, nil
+				return nil, errors.New("x not valid disk")
```
Comment (candysmurf) [inline smart/plugin_test.go]: why is it error out all the time?

File: smart/plugin.go [hunk 2]
```diff
@@ -213,10 +217,11 @@ func (sc *SmartCollector) CollectMetrics(mts []plugin.MetricType) ([]plugin.Metr
 			}
 		} else {
 			// Single disk requested
-
 			collected, result, err := sc.DiskMetrics(ns, t, disk, attribute_path, buffered_results, errs)
 			if err != nil {
-				sc.logger.Error(fmt.Sprintf("Error collecting SMART %s data on %s disk: %#+v", attribute_path, disk, err))
+				eStr := fmt.Sprintf("Error collecting SMART %s data on %s disk: %#+v", attribute_path, disk, err)
+				sc.logger.Error(eStr)
```
Comment (candysmurf) [inline smart/plugin.go]: logging single error here but on the line 235, it logs all the errors. It seems that duplicated errors being logged?

File: smart/plugin.go [hunk 3]
```diff
@@ -203,7 +205,9 @@ func (sc *SmartCollector) CollectMetrics(mts []plugin.MetricType) ([]plugin.Metr
 			for _, dev := range devices {
 				collected, result, err := sc.DiskMetrics(ns, t, dev, attribute_path, buffered_results, errs)
 				if err != nil {
-					sc.logger.Error(fmt.Sprintf("Error collecting SMART %s data on %s disk: %#+v", attribute_path, dev, err))
+					eStr := fmt.Sprintf("Error collecting SMART %s data on %s disk: %#+v", attribute_path, dev, err)
```
Comment (candysmurf) [inline smart/plugin.go]: Same as the comment below. It seems that it logs single error then append to log all errors together. We only need one way to log errors. I think. If the single error is more readable, stick to that.

Comment (obourdon) [inline smart/plugin.go]: Agreed, new PR submitted

File: smart/plugin_test.go (hunk 1 above)
Comment (obourdon) [inline smart/plugin_test.go]: In fact as the next snippet was removed in a previous commit, I had to adapt the code so that the tests pass

**Comment by candysmurf** [comment]
LGTM

**Pull request merged**